docs/DeveloperGuide.md

```diff
@@ -1,12 +1,220 @@
 # Developer Guide
 
 ## Acknowledgements
+Our project uses external libraries and services from:
+1. Land Transport Authority DataMall API Service ([link](https://datamall.lta.gov.sg/content/datamall/en.html)).
+2. Jackson JSON Parser ([link](https://fasterxml.github.io/jackson-core/javadoc/2.8/com/fasterxml/jackson/core/JsonParser.html))
+
+## Design 
+
+### Architecture Level
+
+### Component Level
+
+#### Model Component
+![Model Class Diagram](images/ModelClassDiagram.png)
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The model component consists of a `CarparkList` (and `CarparkFilteredList`) class that contains
+instances of `Carpark`, under the package `seedu.data`. 
+
+This component: 
+* Stores all carpark data (all `Carpark` objects), contained in a `CarparkList` object.
+* Stores filtered carpark data to be used in other components (in `CarparkFilteredList`).
+* Contains methods for selecting a `Carpark` object based on a unique code (for the `find` command) as well
+as filtering by a substring or set of substrings (`search` command).
+* Is independent of other components except the API component, which is used to generate it.
+* Group objects with the same code by enum `LotType` (Car, Motorcycle, Heavy Vehicle) and places them in a HashMap 
+for easy access. For example: Three `carpark` objects may have the same unique carpark code as they are the same 
+carpark, but contain available lot information for different types of lot. These three objects will be grouped under 
+one object with the HashMap `allAvailableLots` containing a breakdown of lots by type.
 
-## Design & implementation
+Note: The `Carpark` class contain many getters, setters and annotations to be used with the `jackson` module. 
+See [`FileLoader`](#FileLoader) for more information.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### API Component
+![API Class Diagram](images/ApiClassDiagram.png)
+
```

Comment:
the chunk of text is hard to grasp/understand. perhaps, you can reduce the number of words/simplify the explanation